pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -451,53 +472,147 @@ func (s *nsmgrSuite) Test_PassThroughRemoteUsecase() {
 
 func (s *nsmgrSuite) Test_PassThroughLocalUsecase() {
 	t := s.T()
-	const nsesCount = 7
+	const nsesCount = 5
+
+	labels := []string{"a", "b", "c", "d", "e"}
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
 
+	nsReg := linearNS(labels)
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
 	var nsesReg [nsesCount]*registry.NetworkServiceEndpoint
 	for i := 0; i < nsesCount; i++ {
+		nsesReg[i] = &registry.NetworkServiceEndpoint{
+			Name:                fmt.Sprintf("endpoint-%v", labels[i]),
+			NetworkServiceNames: []string{nsReg.GetName()},
+			NetworkServiceLabels: map[string]*registry.NetworkServiceLabels{
+				nsReg.GetName(): {
+					Labels: map[string]string{
+						step: labels[i],
+					},
+				},
+			},
+		}
+
 		var additionalFunctionality []networkservice.NetworkServiceServer
-		if i != 0 {
+		if i != nsesCount-1 {
 			additionalFunctionality = []networkservice.NetworkServiceServer{
 				chain.NewNetworkServiceServer(
 					clienturl.NewServer(s.domain.Nodes[0].NSMgr.URL),
 					connect.NewServer(ctx,
-						sandbox.NewCrossConnectClientFactory(sandbox.GenerateTestToken,
-							newPassTroughClient(fmt.Sprintf("my-service-remote-%v", i-1)),
-							kernel.NewClient()),
+						client.NewClientFactory(
+							client.WithName(fmt.Sprintf("endpoint-client-%v", labels[i])),
+							client.WithAdditionalFunctionality(
+								mechanismtranslation.NewClient(),
+								injectlabels.NewClient(nsesReg[i].NetworkServiceLabels[nsReg.Name].Labels),
+								kernel.NewClient(),
+							),
+						),
 						connect.WithDialTimeout(sandbox.DialTimeout),
 						connect.WithDialOptions(sandbox.DefaultDialOptions(sandbox.GenerateTestToken)...),
 					),
 				),
 			}
 		}
 
-		nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, &registry.NetworkService{
-			Name: fmt.Sprintf("my-service-remote-%v", i),
-		})
+		_, err = s.domain.Nodes[0].NewEndpoint(ctx, nsesReg[i], sandbox.GenerateTestToken, additionalFunctionality...)
+		require.NoError(t, err)
+	}
+
+	nsc := s.domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
+
+	request := defaultRequest(nsReg.Name)
+
+	conn, err := nsc.Request(ctx, request)
+	require.NoError(t, err)
+	require.NotNil(t, conn)
+
+	// Path length to first endpoint is 5
+	// Path length from NSE client to other local endpoint is 5
+	require.Equal(t, 5*(nsesCount-1)+5, len(conn.Path.PathSegments))
+
+	// Refresh
+	conn, err = nsc.Request(ctx, request)
+	require.NoError(t, err)
+
+	// Close
+	_, err = nsc.Close(ctx, conn)
+	require.NoError(t, err)
+
+	// Endpoint unregister
+	for i := 0; i < len(nsesReg); i++ {
+		_, err = s.domain.Nodes[0].EndpointRegistryClient.Unregister(ctx, nsesReg[i])
 		require.NoError(t, err)
+	}
+}
 
+func (s *nsmgrSuite) Test_PassThroughLocalUsecaseMultiLabel() {
+	t := s.T()
+	const nsesCount = 9
```

Comment:
When you are using arrays of labels for endpoins, you can remove `nsesCount` and use `len(labels)` instead.

Reply:
it is used for array 'nsesReg' for initialization

Reply:
You can create that array dynamically.
Main point is to not being forced to manually sync `nsesCount` value each time you adjust labels array.